### PR TITLE
[ticket/13771] Allow AJAX errors to support exceptions messages

### PR DESCRIPTION
**Pull request by VSEphpbb**
https://tracker.phpbb.com/browse/PHPBB3-13771
PHPBB3-13771

### CODE REVIEW

File: phpBB/assets/javascript/core.js [hunk 1]
```diff
@@ -263,7 +263,11 @@ phpbb.ajaxify = function(options) {
 			}
 			phpbb.clearLoadingTimeout();
 			var errorText = false;
-			if (typeof errorThrown === 'string' && errorThrown.length > 0) {
+			var responseText = $.parseJSON(jqXHR.responseText);
```
Comment (Nicofuma) [inline phpBB/assets/javascript/core.js]: if responseText isn't a valid json object what happen?

@callumacrae @prototech

Comment (MGaetan89) [inline phpBB/assets/javascript/core.js]: [From the doc](http://api.jquery.com/jQuery.parseJSON/):

> Passing in a malformed JSON string results in a JavaScript exception being thrown.

So I think the exception should be handled here somehow.

Comment (callumacrae) [inline phpBB/assets/javascript/core.js]: Yep, if it's an error handler it's pretty likely that this isn't JSON at all (for example, a 500 error). It should be wrapped in a try-catch block, and `JSON.parse` should be used, not `$.parseJSON`—phpBB doesn't support IE7!

**Comment by VSEphpbb** [comment]
Updated

**Comment by VSEphpbb** [comment]
bump

**Comment by Nicofuma** [comment]
@callumacrae

**Comment by callumacrae** [comment]
lgtm

**Pull request merged**